File: DDCharacterCreator/app/src/main/java/com/example/ddcharactercreator/Character.java
package com.example.ddcharactercreator;

import androidx.annotation.NonNull;
import androidx.room.Entity;
import androidx.room.Ignore;
import androidx.room.PrimaryKey;
import java.io.Serializable;
import java.util.ArrayList;

@Entity(tableName = "characterList")
public class Character implements Serializable {

    private int mLevel;
    private String mRace;
    private String mCharacterClass;
    private String mAlignment;
    @PrimaryKey
    @NonNull
    private String mName;
    private ArrayList<Integer> mStatValues;
    private ArrayList<String> mProficiencyChoices;
    private ArrayList<String> mInventoryList;
    private ArrayList<Integer> mCurrency;
    //add subclass to MainActivity character list?? (Level 4 school of Evocation Elf Wizard?)
    private String mSubclass;
    private ArrayList<Spell> mSpellsKnown;
    private ArrayList<String> mSpellSlotsClicked;

    @Ignore
    public Character(){
    }

    @Ignore
    public Character(int level, String race, String characterClass, String alignment, String name, ArrayList<Integer> statValues){
        mLevel = level;
        mRace = race;
        mCharacterClass = characterClass;
        mAlignment = alignment;
        mName = name;
        mStatValues = statValues;
    }

    @Ignore
    public Character(int level, String race, String characterClass, String alignment, String name, ArrayList<Integer> statValues, ArrayList<String> proficiencyChoices, ArrayList<String> inventoryList, ArrayList<Integer> currency){
        mLevel = level;
        mRace = race;
        mCharacterClass = characterClass;
        mAlignment = alignment;
        mName = name;
        mStatValues = statValues;
        mProficiencyChoices = proficiencyChoices;
        mInventoryList = inventoryList;
        mCurrency = currency;
    }

    public Character(int level, String race, String characterClass, String alignment, String name, ArrayList<Integer> statValues,
                     ArrayList<String> proficiencyChoices, ArrayList<String> inventoryList, ArrayList<Integer> currency,
                     String subclass, ArrayList<Spell> spellsKnown, ArrayList<String> spellSlotsClicked){
        mLevel = level;
        mRace = race;
        mCharacterClass = characterClass;
        mAlignment = alignment;
        mName = name;
        mStatValues = statValues;
        mProficiencyChoices = proficiencyChoices;
        mInventoryList = inventoryList;
        mCurrency = currency;
        mSubclass = subclass;
        mSpellsKnown = spellsKnown;
        mSpellSlotsClicked = spellSlotsClicked;
    }

    public int getLevel(){
        return mLevel;
    }
    public String getRace(){
        return mRace;
    }
    public String getCharacterClass(){
        return mCharacterClass;
    }
    public String getAlignment(){
        return mAlignment;
    }
    public String getName(){
        return mName;
    }
    public ArrayList<Integer> getStatValues(){
        return mStatValues;
    }
    public ArrayList<String> getProficiencyChoices(){
        return mProficiencyChoices;
    }
    public ArrayList<String> getInventoryList(){
        return mInventoryList;
    }
    public ArrayList getCurrency(){
        return mCurrency;
    }
    public String getSubclass(){
        return mSubclass;
    }
    public ArrayList<Spell> getSpellsKnown(){
        return mSpellsKnown;
    }
    public ArrayList<String> getSpellSlotsClicked(){
        return mSpellSlotsClicked;
    }
}

File: DDCharacterCreator/app/src/main/java/com/example/ddcharactercreator/Character.kt
package com.example.ddcharactercreator

import androidx.annotation.NonNull
import androidx.room.Entity
import androidx.room.Ignore
import androidx.room.PrimaryKey
import java.io.Serializable
import java.util.*

@Entity(tableName = "characterList")
class Character: Serializable{

    private var mLevel: Int = 0
    private var mRace: String = ""
    private var mCharacterClass: String = ""
    private var mAlignment: String = ""
    @PrimaryKey
    @NonNull
    private var mName: String = ""
    private var mStatValues: ArrayList<Int> = ArrayList<Int>()
    private var mProficiencyChoices: ArrayList<String> = ArrayList<String>()
    private var mInventoryList: ArrayList<String> = ArrayList<String>()
    private var mCurrency: ArrayList<Int> = ArrayList<Int>()
    private var mSubclass: String = ""
    private var mSpellsKnown: ArrayList<Spell> = ArrayList<Spell>()
    private var mSpellSlotsClicked: ArrayList<String> = ArrayList<String>()
    private var mRaceAndClassBonusStats: ArrayList<String> = ArrayList<String>()
    private var mClassBasedBonusStats2: ArrayList<String> = ArrayList<String>()

    @Ignore
    constructor(){
    }

    @Ignore
    constructor(level: Int, race: String, characterClass: String, alignment: String, name: String, statValues: ArrayList<Int>){
        mLevel = level
        mRace = race
        mCharacterClass = characterClass
        mAlignment = alignment
        mName = name
        mStatValues = statValues
    }

    @Ignore
    constructor(level: Int, race: String, characterClass: String, alignment: String, name: String, statValues: ArrayList<Int>, proficiencyChoices: ArrayList<String>,
                inventoryList: ArrayList<String>, currency:ArrayList<Int>){
        mLevel = level
        mRace = race
        mCharacterClass = characterClass
        mAlignment = alignment
        mName = name
        mStatValues = statValues
        mProficiencyChoices = proficiencyChoices
        mInventoryList = inventoryList
        mCurrency = currency
    }

    constructor(level: Int, race: String, characterClass: String, alignment: String, name: String, statValues: ArrayList<Int>, proficiencyChoices: ArrayList<String>,
                inventoryList: ArrayList<String>, currency:ArrayList<Int>, subclass: String, spellsKnown: ArrayList<Spell>, spellSlotsClicked: ArrayList<String>,
                raceAndClassBonusStats: ArrayList<String>, classBasedBonusStats2: ArrayList<String>){
        mLevel = level
        mRace = race
        mCharacterClass = characterClass
        mAlignment = alignment
        mName = name
        mStatValues = statValues
        mProficiencyChoices = proficiencyChoices
        mInventoryList = inventoryList
        mCurrency = currency
        mSubclass = subclass
        mSpellsKnown = spellsKnown
        mSpellSlotsClicked = spellSlotsClicked
        mRaceAndClassBonusStats = raceAndClassBonusStats
        mClassBasedBonusStats2 = classBasedBonusStats2
    }

    fun getLevel(): Int{
        return mLevel
    }
    fun getRace(): String{
        return mRace
    }
    fun getCharacterClass(): String{
        return mCharacterClass
    }
    fun getAlignment(): String{
        return mAlignment
    }
    fun getName(): String{
        return mName
    }
    fun getStatValues(): ArrayList<Int>{
        return mStatValues
    }
    fun getProficiencyChoices(): ArrayList<String>{
        return mProficiencyChoices
    }
    fun getInventoryList(): ArrayList<String>{
        return mInventoryList
    }
    fun getCurrency(): ArrayList<Int>{
        return mCurrency
    }
    fun getSubclass(): String{
        return mSubclass
    }
    fun getSpellsKnown(): ArrayList<Spell>{
        return mSpellsKnown
    }
    fun getSpellSlotsClicked(): ArrayList<String>{
        return mSpellSlotsClicked
    }
    fun getRaceAndClassBonusStats(): ArrayList<String>{
        return mRaceAndClassBonusStats
    }
    fun getClassBasedBonusStats2(): ArrayList<String>{
        return mClassBasedBonusStats2
    }
}